noir-projects/noir-protocol-circuits/crates/types/src/hash.nr

```diff
@@ -151,16 +151,23 @@ pub fn compute_l2_to_l1_hash(
     rollup_version_id: Field,
     chain_id: Field,
 ) -> Field {
-    let mut bytes: [u8; 160] = std::mem::zeroed();
-
-    let inputs =
-        [contract_address.to_field(), rollup_version_id, recipient.to_field(), chain_id, content];
-    for i in 0..5 {
-        // TODO are bytes be in fr.to_buffer() ?
-        let item_bytes: [u8; 32] = inputs[i].to_be_bytes();
-        for j in 0..32 {
-            bytes[32 * i + j] = item_bytes[j];
-        }
+    let contract_address_bytes: [u8; 32] = contract_address.to_field().to_be_bytes();
+    let recipient_bytes: [u8; 20] = recipient.to_be_bytes();
+    let content_bytes: [u8; 32] = content.to_be_bytes();
+    let rollup_version_id_bytes: [u8; 32] = rollup_version_id.to_be_bytes();
+    let chain_id_bytes: [u8; 32] = chain_id.to_be_bytes();
+
+    let mut bytes: [u8; 148] = std::mem::zeroed();
+    for i in 0..32 {
+        bytes[i] = contract_address_bytes[i];
+        bytes[i + 0x20] = rollup_version_id_bytes[i];
```

Comment:
It can be slightly confusing to read when there is a mix of hex and decimal used throughout.